### DISCUSSION TITLE
SwiftPM Bug

### DISCUSSION THREAD

**Discussion by JacksonUtsch** [tgrapperon/swift-dependencies-additions#27] [General]
Not entirely related to this package but caused me troubles in using it.

In an application I'm working on I attempted to import this package but this package and another dependency contain modules with the same names. So the "solution" is to use SwiftPM's `moduleAliases`. This requires `swift-tools-version:5.7` which required me to update Xcode 🙈. Xcode threw me some misleading errors and it took me a while to find out `moduleAliases` has a [bug](https://github.com/apple/swift-package-manager/issues/5762) that causes issues with other dependencies using previous package versions. So ultimately _my solution_ was to rename one of the modules sharing names so they don't.

With the number of modules in this package I thought someone else might run into this eventually.

**Comment by tgrapperon** [comment]
Ah, this is interesting. I've opted to name dependencies as `XXXDependency`, thinking that it would be enough to avoid collisions with user modules. Out of curiosity, what is the conflicting name? As I'm understanding it, if you were able to bump your package to 5.7, `moduleAliases` would have worked, right?
Many thanks for reporting the issue.

**Reply by JacksonUtsch** [reply]
> Out of curiosity, what is the conflicting name?

UserDefaultsDependency 😂 This is a module I made previously.

> As I'm understanding it, if you were able to bump your package to 5.7, moduleAliases would have worked, right?

Sadly no. It seems **all** other dependencies would require the `5.7` package version as well. These other packages start giving errors like `product x required by package y target z not found in package x`. This seems odd but is what I am seeing.
